### PR TITLE
feat(docs): Website landing page

### PR DESCRIPTION
**Pull request by cfraz89**
Creating the PR early so as to have a netlify preview build live. 

This PR is hosting the WIP overhaul of the site theme and landing page.

### CODE REVIEW

File: website/src/components/blog.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+export enum ChipColor {
+  blue = "blue-chip",
+  green = "green-chip",
+  yellow = "yellow-chip",
+  purple = "purple-chip",
+}
+
+type Props = {
```
Comment (sam-goodwin) [inline website/src/components/blog.tsx:8]: Favor interface over type aliases for static types.

File: website/src/components/features.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+              languages.
+            </p>
+          </div>
+          <div className="mt-10">
+            <h5 className="m-0">Automated IAM policies</h5>
+            <p className="body1 text-functionless-medium dark:text-functionless-dark-medium mt-2">
+              The best security auditor is no auditor - our compiler derives
+              minimal IAM Policies from your code.
+            </p>
+          </div>
```
Comment (sam-goodwin) [inline website/src/components/features.tsx:36]: Should we create a component for feature

**Review by sam-goodwin — COMMENTED** [review]
Looks like build failed ?

File: website/src/components/features.tsx (hunk 2 above)
Comment (cfraz89) [reply website/src/components/features.tsx:36]: definitely

File: website/src/components/blog.tsx (hunk 1 above)
Comment (cfraz89) [reply website/src/components/blog.tsx:8]: sure

**Comment by cfraz89** [comment]
> Looks like build failed ?

yeah no idea why right now, will have to mess around with it

**Comment by thantos** [comment]
nit: you can mark this a "draft" instead of WIP.

image

File: website/src/content/home/hero.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export const announcement = {
+  title: "v0.22.28 Released!",
+  href: "https://github.com/functionless/functionless/releases",
+};
+
+export const title = "Code the Cloud";
```
Comment (sam-goodwin) [inline website/src/content/home/hero.ts:6]: Let's change that to "Code-first, Cloud-native`

File: website/src/content/home/features/features.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const title = "FEATURES";
+export const subtitle =
+  "Build Reliable and Scalable systems with Smart Resources.";
```
Comment (sam-goodwin) [inline website/src/content/home/features/features.ts:3]: This is way too long, especially in mobile. I wonder if we should trash it?

Comment (sam-goodwin) [reply website/src/content/home/features/features.ts:3]: It doesn't add much value - just have the `FEATURES` small piece of text and then dive straight into the sections.

**Review by sam-goodwin — COMMENTED** [review]
Looking great, good work! We have some copy tweaks to make but mostly looking ready to go.

One blocker i've noticed is that the scrolling can be jumpy and unintuitive. While scrolling, it seems to rubberband.

**Review by sam-goodwin — APPROVED** [review]

File: website/src/components/home/features/asides/tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+              <Code
+                animate={visible}
+                fileName="functionless.ts"
+                language="typescript"
+                introDelayMs={250}
+              >
+                <FunctionlessTableFunction />
+              </Code>
+            )}
+          </VisibilityWindow>
```
Comment (thantos) [inline website/src/components/home/features/asides/tabs.tsx:50]: remove unused code.

File: website/src/components/home/subscribe.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+              {
+                name: "firstName",
+                value: "N/A",
+                objectTypeId: HubSpotTypeId.SingleLineText,
+              },
+              {
+                name: "lastName",
+                value: "N/A",
+                objectTypeId: HubSpotTypeId.SingleLineText,
+              },
```
Comment (thantos) [inline website/src/components/home/subscribe.tsx:54]: no longer mandatory

File: website/src/content/home/features/code/output.mdx [hunk 7]
```diff
@@ -0,0 +1,11 @@
+```json
+// output
+{
+  "Item": {
+    "id": {
+      "S": "user-id",
+    },
+    ..
```
Comment (thantos) [inline website/src/content/home/features/code/output.mdx:8]: missing a dot?

File: website/src/content/home/features/code/table-function.mdx [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+const getUser = new Function(this, "getUser", async (userId: string) => {
+  return $AWS.DynamoDB.GetItem({
+    Table: userTable,
+    Key: {
+      id: {
+        S: userId,
+      },
+    },
+  });
```
Comment (thantos) [inline website/src/content/home/features/code/table-function.mdx:12]: update to new dynamo syntax

File: website/src/content/home/features/snippets/appsync-resolver.mdx [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+export default AppsyncResolver<{ userId: string }, User>(
+  {
+    typeName: "Query",
+    fieldName: "getUser",
+  },
+  async ($context) =>
+    userTable.appsync.getItem({
+      id: $util.dynamodb.toDynamoDB($context.arguments.userId),
+    })
```
Comment (thantos) [inline website/src/content/home/features/snippets/appsync-resolver.mdx:12]: update to new api

File: website/src/content/home/features/snippets/get-user.mdx [hunk 10]
```diff
@@ -0,0 +1,11 @@
+```ts
+import { users } from "./user-table";
+
+export default Function(async (name: string) => {
+  return users.getItem({
+    key: {
+      name,
+    },
+  });
```
Comment (thantos) [inline website/src/content/home/features/snippets/get-user.mdx:9]: Update to new api

File: website/src/content/home/nav.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+export const main = [
+  {
+    to: "/docs/what-is-functionless",
+    title: "Docs",
+  },
+  // { to: "/blog", title: "Blog" },
+  // {
+  //   to: "/team",
+  //   title: "Team",
+  // },
```
Comment (thantos) [inline website/src/content/home/nav.ts:23]: remove?

**Review by thantos — APPROVED** [review]

**Pull request merged**